### PR TITLE
dictGet and dictGetOrDefault for Decimals

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- New Feature

Short description (up to few sentences):
Implement dictGet and dictGetOrDefault functions for Decimal types. It extracts type arguments (precision and scale) from dictionary attributes described in XML.

### CODE REVIEW

File: dbms/src/Core/Types.h [hunk 1]
```diff
@@ -170,6 +170,46 @@ template <> struct NativeType<Decimal32> { using Type = Int32; };
 template <> struct NativeType<Decimal64> { using Type = Int64; };
 template <> struct NativeType<Decimal128> { using Type = Int128; };
 
+inline const char * getTypeName(TypeIndex idx)
+{
+    switch (idx)
```
Comment (alexey-milovidov) [inline dbms/src/Core/Types.h]: BTW can be done with X macro.

File: dbms/src/Dictionaries/FlatDictionary.cpp [hunk 2]
```diff
@@ -701,10 +701,10 @@ void FlatDictionary::setAttributeValue(Attribute & attribute, const Key id, cons
             break;
 
         case AttributeUnderlyingType::Decimal32:
-            setAttributeValueImpl<Decimal32>(attribute, id, value.get<Decimal128>());
+            setAttributeValueImpl<Decimal32>(attribute, id, value.get<Decimal32>());
             break;
         case AttributeUnderlyingType::Decimal64:
-            setAttributeValueImpl<Decimal64>(attribute, id, value.get<Decimal128>());
+            setAttributeValueImpl<Decimal64>(attribute, id, value.get<Decimal64>());
```
Comment (alexey-milovidov) [inline dbms/src/Dictionaries/FlatDictionary.cpp]: A perfect case for PVS-Studio.

**Comment by alexey-milovidov** [comment]
The only remaining is the integration test.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by alexey-milovidov** [comment]
`clickhouse-test-server` script is not used in CI.
It is @proller's own script.

**Comment by alesapin** [comment]
Temporary we can add this `.xml` files to docker images. After DDL for dictionaries will be merged we can write functional tests for dictionaries in simple way.

**Pull request merged**